### DISCUSSION TITLE
Can RadioButtons use the keyboard?

### DISCUSSION THREAD

**Discussion by voidware** [ocornut/imgui#3944] [Old/Archived discussions]
I'd like to give the keyboard focus to a set of radio buttons, but how.

Currently, i'm using this hack:

```
    ImGui::SetFocusID(ImGui::GetID(name), ImGui::GetCurrentWindow()); 
```

But it's not nice.

Also, does it really use the text on the button as the ID? What happens if more than one button have the same text?

**Comment by ocornut** [comment]
> But it's not nice.

What precisely is not nice about it?

If submitted after said item, you can also use `ImGui::GetItemID()` from imgui_internal.h. to retrieve the last ID.

Note that your subject line "Can RadioButtons use the keyboard" has nothing to do with your "SetFocusID()" example so it looks like you are asking about two things? You can enable Keyboard Navigation to provide full keyboard controls.

> Also, does it really use the text on the button as the ID? What happens if more than one button have the same text?

Read the FAQ.

**Reply by voidware** [reply]
Only that you have to include internal.h

`GetItemID` like it! that saves the lookup. will fix.

**Reply by voidware** [reply]
I thought i did enable navigation;

```
    io.ConfigFlags |= ImGuiConfigFlags_NavEnableKeyboard;     // Enable Keyboard Controls
```

Can that "tab" (or however) to the radio buttons without having to manually set focus?

**Reply by ocornut** [reply]
I think down the line we will provide more functions to programmatically manipulate focus and activate item and then we can provide higher-level helpers, and I suppose we could add a `SetItemFocus()`.

It's mostly a matter of tidying up the mess that is the old SetKeyboardFocusHere() and old Tabbing system and unifying all of it into nice words.

**Reply by voidware** [reply]
Thanks. I'll run with what i have for now. I just wondered if i missed something obvious.
